Migrate build logic to Kotlin DSL

3flex opened the pull request:
Closes #186 

The diff from GitHub makes it hard to review, but the commit history should be clear.

Goooler commented on build.gradle.kts:
```diff
@@ -0,0 +1,128 @@
+}
+
+githubRelease {
+    token(System.getenv("CCUD_GIT_TOKEN") ?: "")
```
```suggestion
    token(providers.environmentVariable("CCUD_GIT_TOKEN").orNull.orEmpty())
```

clayburn commented on settings.gradle.kts:
```diff
@@ -0,0 +1,33 @@
+plugins {
+    id("com.gradle.develocity") version "3.17"
+    id("com.gradle.common-custom-user-data-gradle-plugin") version "2"
```
```suggestion
    id("com.gradle.common-custom-user-data-gradle-plugin") version "2.0"
```

clayburn reviewed (changes requested):
Thanks for the contribution @3flex! I made one suggestion. If you take that and fix the merge conflict, we'd be happy to accept this.

3flex commented:
Fixed - I've also updated README in the last commit as there were a few references (and a link) to settings.gradle. Please check that before merging, thanks!
